gen3-integration-tests/services/graph.py

```diff
@@ -235,3 +253,193 @@ def query_node_count(self, node_name: str) -> dict:
         """
         query = f'{{ _{node_name}_count (project_id: "{self.project_id}") }}'
         return self.graphql_query(query)
+
+    def get_path_with_file_node(self, path_to_file=False, file_node=False):
+        """
+        Get all nodes except node with _file category or get node with _file category
+
+        Args:
+            path_to_file: returns all nodes except for node with _file category
+            file_node : returns the node with _file category
+        """
+        all_nodes = copy.deepcopy(self.test_records)
+        file_node_name = ""
+        # Get file node name with _file category
+        for node_name in self.submission_order:
+            if "_file" in all_nodes[node_name].category:
+                file_node_name = node_name
+        file_node_item = all_nodes[file_node_name]
+        del all_nodes[file_node_name]
+        # Return all nodes except for the node with _file category recieved above
+        if path_to_file:
+            return all_nodes
+        # Return node with _file category recieved above
+        if file_node:
+            return file_node_item
+
+    def get_field_of_type(self, node: dict, field_type: object) -> str:
+        """
+        Gets the field of desired data type
+
+        Args:
+            field_type: python object like str, int, dict, etc.
+        """
+        for key, val in node.props.items():
+            if isinstance(val, field_type):
+                return key
+
+    def get_did_from_file_id(self, guid: str) -> str:
+        """
+        Returns the did/indexd_guid value for file node
+        Args:
+            guid: node's entities based id
+        """
+        response = requests.get(
+            url=pytest.root_url
+            + self.BASE_ADD_ENDPOINT
+            + "{}/{}/export?ids={}&format=json".format(
+                self.program_name, self.project_code, guid
+            ),
+            auth=self.auth,
+        )
+        return response.json()[0]["object_id"]
+
+    def submit_graph_and_file_metadata(
```

Comment:
this logic from the old tests needs to be cleaned up. the `submit_new_record` function i wrote in this file was an attempt at simplifying and replacing this logic. Let's talk about it tomorrow